plugins/extensions/plugins/jsonSerialization.h

```diff
@@ -415,3 +424,52 @@ void init(brayns::AnimationParameters* a, ObjectHandler* h)
     h->set_flags(Flags::DisallowUnknownKey);
 }
 }
+
+namespace brayns
+{
+template <class T>
+inline std::string to_json(const T& obj)
+{
+    return staticjson::to_pretty_json_string(obj);
+}
+template <>
+inline std::string to_json(const Engine::Progress& obj)
+{
+    std::lock_guard<std::mutex> lock(obj.mutex);
+    return staticjson::to_pretty_json_string(obj);
+}
+template <>
+inline std::string to_json(const Version& obj)
+{
+    return obj.toJSON();
+}
+template <class T, class F>
+inline bool from_json(T& obj, const std::string& json, F postUpdateFunc = [] {})
+{
+    staticjson::ParseStatus status;
+    const auto success =
+        staticjson::from_json_string(json.c_str(), &obj, &status);
+    if (success)
+    {
+        obj.markModified();
+        if (std::function<void(T&)>(postUpdateFunc))
+            postUpdateFunc(obj);
+    }
+    else
+        BRAYNS_ERROR << status.description() << std::endl;
+    return success;
+}
+}
+
+// for rockets::jsonrpc
```

Comment:
is there a reason to place these two functions separately, outside of the brayns namespace?

Reply:
yes, to make it compile :) rockets expects them to be free functions within no namespace

Reply:
Oh ok, but then the same does not apply to the other from/to_json above..?

Reply:
Interestingly not... I move them together.